code/espurna/config/general.h

```diff
@@ -626,14 +626,48 @@
 // -----------------------------------------------------------------------------
 
 #ifndef OTA_PORT
-#define OTA_PORT                    8266        // OTA port
+#define OTA_PORT                    8266        // Port for ArduinoOTA
 #endif
 
 #ifndef OTA_MQTT_SUPPORT
-#define OTA_MQTT_SUPPORT           0            // No support by default
+#define OTA_MQTT_SUPPORT            0           // Listen for HTTP(s) URLs at '<root topic>/ota'. Depends on OTA_CLIENT
 #endif
 
-#define OTA_GITHUB_FP               "D7:9F:07:61:10:B3:92:93:E3:49:AC:89:84:5B:03:80:C1:9E:2F:8B"
+#ifndef OTA_ARDUINOOTA_SUPPORT
+#define OTA_ARDUINOOTA_SUPPORT      1            // Support ArduinoOTA by default
+                                                 // Implicitly depends on ESP8266mDNS library, thus increasing firmware size
+#endif
+
+#ifndef OTA_CLIENT
+#define OTA_CLIENT                  OTA_CLIENT_ASYNCTCP     // Terminal / MQTT OTA support
+                                                            // OTA_CLIENT_ASYNCTCP   (ESPAsyncTCP library)
+                                                            // OTA_CLIENT_HTTPUPDATE (Arduino Core library)
+#endif
+
+#define OTA_GITHUB_FP               "CA:06:F5:6B:25:8B:7A:0D:4F:2B:05:47:09:39:47:86:51:15:19:84"
+
+#ifndef OTA_FINGERPRINT
+#define OTA_FINGERPRINT             OTA_GITHUB_FP
+#endif
+
+#ifndef OTA_SSL_CLIENT_INCLUDE_CA
+#define OTA_SSL_CLIENT_INCLUDE_CA        0               // Use user-provided CA (only PROGMEM option is supported):
+                                                         // const char _ota_client_http_update_ca[] PROGMEM = "...";
+                                                         // By default, use Letsencrypt ISRG Root X1
+#endif
+
+// -----------------------------------------------------------------------------
+// SSL Client
+// -----------------------------------------------------------------------------
+
+#ifndef SSL_CLIENT
+#define SSL_CLIENT                          SSL_CLIENT_NONE     // What variant of WiFiClient to use (no SSL support by default)
```

Comment:
I think it would be good to mention the option here, so: ``SSL_CLIENT_NONE``, ``SSL_CLIENT_AXTLS`` and ``SSL_CLIENT_BEARSSL``. Also maybe do some auto detection based on the core version?

Reply:
Autodetection? Cipher problem that was kind of recent one (stack thunk size increase) and can be mitigated locally by using hand-made cipher list. And that affects 2.5.0-2.5.2. It kind of tricky to define what works, so maybe it is better to leave this alone.
I will add a comment about possible clients options + Core from which they work / become deprecated etc